Add schematic details and links to config page (#27272)
angular/angular: aio/content/guide/workspace-config.md

## Overall JSON structure

> At the top level of `angular.json`, a few properties configure the workspace, and a `projects` section contains the remaining per-project configuration options. CLI defaults set at the workspace level can be overridden by defaults set at the project level, and defaults set at the project level can be overridden on the command line.

Reviewer: Can we add something to the effect of "These are the top level workspace properties:" here?

## Overall JSON structure

At the top level of `angular.json`, a few properties configure the workspace, and a `projects` section contains the remaining per-project configuration options. CLI defaults set at the workspace level can be overridden by defaults set at the project level, and defaults set at the project level can be overridden on the command line.

> The following properties at the top level of the file configure the workspace.

Reviewer: ```suggestion
The following are properties at the top level of the workspace configuration file:
```

> The following properties are at the top level of the file configure the workspace.

Reviewer: My whole suggestion was not accepted for this line. It still has some grammar issues even after that update.

Author: Oops, fixing typo. This line introduces the list of top-level properties: 
"The following properties at the top level of the file configure the workspace."
 What do you think is wrong with it?

Reviewer: Oh I think that I see how you want it to read... perhaps this would make that more clear to the reader:
"The following properties, at the top level of the file, configure the workspace."

Author: You're right. Done.

| `prefix`        | A string that Angular prepends to generated selectors. Can be customized to identify an app or feature area. |
| `schematics`    | A collection of schematics that customize the `ng generate` sub-command options for this project. See [Generation schematics](#schematics) below.  |
| `architect`     | Configuration defaults for Architect builder targets for this project. |

{@a schematics}
## Generation schematics

Angular generation schematics are instructions for modifying a project by adding files to it or modifying existing files. Individual schematics are collected in packages. Specify a schematic in the format `schematic-package:schematic-name`.

The JSON schemas for the default schematics used by the CLI to generate project components are collected in the [`@schematics` package](https://github.com/angular/angular-cli/blob/v6.0.0-rc.8/packages/%40angular/cli/lib/config/schema.json).
These schematics configure the options available to the CLI for generating the following `ng generate` sub-commands:

> * [component](https://github.com/angular/angular-cli/blob/v6.0.0-rc.8/packages/%40angular/cli/lib/config/schema.json#L74-L144)

Reviewer 2: I don't think we should use tags here, at least not for version 6.

But what should we use instead, thoughts?

//cc @hansl

Reviewer 3: The tag is fine. I don't think we should use RC tags, only stable ones.

Is there a reason this points to 6.0 instead of 7?

Author: I'm not sure we should even have these individual links - Are they useful?
They were from the wiki page  -- one of our requirements was to make sure all information got transferred from the old wiki to the regular aio doc.

| `sourceRoot`    | The root folder for this project's source files. |
| `projectType`   | One of "application" or "library". An application can run independently in a browser, while a library cannot. Both an app and its e2e test app are of type "application".|
| `prefix`        | A string that Angular prepends to generated selectors. Can be customized to identify an app or feature area. |
| `schematics`    | A set of schematics that customize the `ng generate` sub-command option defaults for this project. See [Generation schematics](#schematics) below.  |
| `architect`     | Configuration defaults for Architect builder targets for this project. |

{@a schematics}
## Generation schematics

> Angular generation [schematics](guide/glossary#schematic) are instructions for modifying a project by adding files to it or modifying existing files.

Reviewer: It's possible for schematics to remove files from a project as well.

> Angular generation [schematics](guide/glossary#schematic) are instructions for modifying a project by adding files to it or modifying existing files.

Reviewer: ```suggestion
Angular generation [schematics](guide/glossary#schematic) are instructions for modifying a project by adding files or modifying existing files.
```

> Individual schematics for Angular CLI `ng generate` sub-commands are collected in the package `@angular`.

Reviewer: ```suggestion
Individual schematics for the default Angular CLI `ng generate` sub-commands are collected in the package `@angular`.
```

> Specify the schematic named for a subcommand in the format `schematic-package:schematic-name`;

Reviewer: ```suggestion
Specify the schematic name for a subcommand in the format `schematic-package:schematic-name`;
```

for example, the schematic for generating a component is `@angular:component`.

> The JSON schemas for the default schematics used by the CLI to generate projects and parts of projects are collected in the package [`@schematics/angular`](https://raw.githubusercontent.com/angular/angular-cli/v7.0.0/packages/schematics/angular/application/schema.json).

Reviewer: We should not link to the raw content here. https://github.com/angular/angular-cli/blob/7.0.x/packages/schematics/angular/application/schema.json goes to the latest `7.0.x` version of this file.

Reviewer: ```suggestion
The JSON schemas for the default schematics used by the CLI to generate projects and parts of projects are collected in the package [`@schematics/angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/schematics/angular/application/schema.json).
```

The schema describes the options available to the CLI for each of the `ng generate` sub-commands, as shown in the `--help` output.

The fields given in the schema correspond to the allowed argument values and defaults for the CLI sub-command options.
You can update your workspace schema file to set a different default for a sub-command option.

{@a architect}

## Project tool configuration options

> Architect is the tool that the CLI uses to perform complex tasks such as compilation and test running, according to provided configurations.

Reviewer: ```suggestion
Architect is the tool that the CLI uses to perform complex tasks, such as compilation and test running, according to provided configurations.
```

| PROPERTY | DESCRIPTION |
| :-------------- | :---------------------------- |
| `builder`       | The npm package for the build tool used to create this target. The default is `@angular-devkit/build-angular:browser`, which uses the [webpack](https://webpack.js.org/) package bundler. |
| `options`       | This section contains default schematics for target options, used when no named alternative configuration is specified. See [Default target schematics](#default-target-schematics) below. |
| `configurations`| This section defines and names alternative configurations for different intended destinations. It contains a section for each named configuration, which sets the default options for that intended environment. See [Alternate build configurations](#build-configs) below. |

{@a default-target-schematics}

### Default target schematics

Angular defines default schematics for use with the Architect tool and `ng run` command. The schematics provide instructions for using a particular tool to perform a complex operation.

> These schematics are collected in the [`@angular-devkit/build-angular` package](https://github.com/angular/angular-cli/blob/v6.0.0-rc.8/packages/%40angular/cli/lib/config/schema.json). They configure options for the following Architect targets:

Reviewer 2: Nit: `package` should not be part of the link.

| PROPERTY | DESCRIPTION |
| :-------------- | :---------------------------- |
| `builder`       | The npm package for the build tool used to create this target. The default is `@angular-devkit/build-angular:browser`, which uses the [webpack](https://webpack.js.org/) package bundler. |
| `options`       | This section contains default schematics for target options, used when no named alternative configuration is specified. See [Default target schematics](#default-target-schematics) below. |
| `configurations`| This section defines and names alternative configurations for different intended destinations. It contains a section for each named configuration, which sets the default options for that intended environment. See [Alternate build configurations](#build-configs) below. |

{@a default-target-schematics}

### Default target schematics

Angular defines default schematics for use with the Architect tool and `ng run` command.
The schematics provide instructions for using a particular tool to perform a complex operation.

> The JSON schemas that the define the options and defaults for each of these schematics are collected in the [`@angular-devkit/build-angular`](https://github.com/angular/angular-cli/blob/v7.1.3/packages/angular/cli/lib/config/schema.json) package. The schemas configure options for the following Architect targets:

Reviewer: This should probably link to https://github.com/angular/angular-cli/blob/7.0.x/packages/angular/cli/lib/config/schema.json unless we want to update this one and the above to go to `7.3.x` instead of `7.0.x`.

It's probably best to link to the stable major release if there aren't important changes in the minors as we likely don't want to update the doc links for each minor release.

Reviewer: ```suggestion
The JSON schemas that the define the options and defaults for each of these schematics are collected in the [`@angular-devkit/build-angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/angular/cli/lib/config/schema.json) package. The schemas configure options for the following Architect targets:
```

| PROPERTY | DESCRIPTION |
| :-------------- | :---------------------------- |
| `builder`       | The npm package for the build tool used to create this target. The default is `@angular-devkit/build-angular:browser`, which uses the [webpack](https://webpack.js.org/) package bundler. |

> | `options`       | This section contains default schematics for target options, used when no named alternative configuration is specified. See [Default target schematics](#default-target-schematics) below. |

Reviewer: This doesn't quite seem accurate to me. Here's a common Architect `build` object like Angular `7.2.10` or `8.0.0-beta.7` generates
```js
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/web",
            "index": "projects/web/src/index.html",
            "main": "projects/web/src/main.ts",
            "polyfills": "projects/web/src/polyfills.ts",
            "tsConfig": "projects/web/tsconfig.app.json",
            "assets": [
              "projects/web/src/favicon.ico",
              "projects/web/src/assets"
            ],
            "styles": [
              "projects/web/src/styles.scss"
            ],
            "scripts": [],
            "es5BrowserSupport": true
          },
```
Categorizing `outputPath`, `index`, `main`, `polyfills`, etc as "default schematics" doesn't seem right. They do seem to be "default build target options" though.

| `configurations`| This section defines and names alternative configurations for different intended destinations. It contains a section for each named configuration, which sets the default options for that intended environment. See [Alternate build configurations](#build-configs) below. |

{@a default-target-schematics}

> ### Default target schematics

Reviewer: This section seems to be describing "Default builders" or "Default build targets", not "target schematics".

| PROPERTY | DESCRIPTION |
| :-------------- | :---------------------------- |
| `builder`       | The npm package for the build tool used to create this target. The default is `@angular-devkit/build-angular:browser`, which uses the [webpack](https://webpack.js.org/) package bundler. |
| `options`       | This section contains default build target options, used when no named alternative configuration is specified. See [Default build targets](#default-build-targets) below. |
| `configurations`| This section defines and names alternative configurations for different intended destinations. It contains a section for each named configuration, which sets the default options for that intended environment. See [Alternate build configurations](#build-configs) below. |

{@a default-build-targets}

### Default build targets

> Angular defines default schematics for use with the Architect tool and `ng run` command.

Reviewer: ```suggestion
Angular defines default builders for use with the Architect tool and `ng run` command.
```

> The schematics provide instructions for using a particular tool to perform a complex operation.

Reviewer: ```suggestion
The default builders provide implementations using a particular tool to perform a complex operation. Builders may do things like perform builds (using tools like Webpack, ng-packagr, and others), run schematics, execute tooling from a library like tslint, and execute test infrastructure (like Karma and Protractor).
```

Reviewer: I'd like some feedback on why the details in the second part of this suggestion were not accepted 😄

> The JSON schemas that the define the options and defaults for each of these schematics are collected in the [`@angular-devkit/build-angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/angular/cli/lib/config/schema.json) package. The schemas configure options for the following Architect targets:

Reviewer: ```suggestion
The JSON schemas that the define the options and defaults for each of these builders are collected in the [`@angular-devkit/build-angular`](https://github.com/angular/angular-cli/blob/7.0.x/packages/angular/cli/lib/config/schema.json) package. The schemas configure options for the following Architect build targets:
```
